### DISCUSSION TITLE
Budibase db full export faster solution?

### DISCUSSION THREAD

**Discussion by uzemelteteshj** [Budibase/budibase#11218] [Help] [answered]
How can i export full full data from budibase db? if I use plain export it only exports 1000 lines. If I scroll down and select all of them it's fine but when I have 6-7 00 000 rows it's a bit time consuming.

**Comment by ConorWebb96** [comment, marked as answer]
Hey @uzemelteteshj,

I wasn't aware you go scroll down and select all the rows the way you are describing. 

The way which I would suggest would be implementing some form of script using the Budibase public API. You can query the rows in batches and then export each batch of 1000 rows to a CSV, from here this could then be imported into a different Budibase instance.

I've actually put a node project together here which you could use to export your table rows to a CSV here; https://github.com/ConorWebb96/bb-export-table-to-csv. If you have any questions or queries on how to use this let me know.

**Reply by uzemelteteshj** [reply]
It's work thank you!!! A hint would be to include in the description that you can specify a unique link at "fetchRows(`https://customdomain.com/api/public/v1/tables/${tableID}/rows/search`, null)" in index.js and can also be used on a own host system

**Reply by ConorWebb96** [reply]
If you think it would be helpful I'll modify it, to include an env variable for self-host? It would just include the first part of that url you've shown https up to .com

**Reply by uzemelteteshj** [reply]
does not necessarily have to be written to an .env file. Just refer to the github description if budi running self host, you may want to modify the url there.
